File: 1455_Check_If_a_Word_Occurs_As_a_Prefix_of_Any_Word_in_a_Sentence/Solution.cpp
class Solution {
public:
    int isPrefixOfWord(string sentence, string searchWord) {
        int iterator = -1;
        int words = 1;
        
        do {
            iterator++;
            if(sentence.substr(iterator, searchWord.length()) == searchWord){
                return words;
            }
            
            iterator = sentence.find(" ", iterator + 1);
            words++;
            
        } while(iterator < sentence.length() && iterator != string::npos);
        
        return -1;
    }
};

File: 1455_Check_If_a_Word_Occurs_As_a_Prefix_of_Any_Word_in_a_Sentence/Solution.java
class Solution {
    public int isPrefixOfWord(String sentence, String searchWord) {
        int iterator = 0;
        int words = 1;
        
        do {
            if(!((iterator + searchWord.length()) > sentence.length()) &&
               sentence.substring(iterator, iterator + searchWord.length()).equals(searchWord)) {
                return words;
            }
            iterator = sentence.indexOf(" ", iterator+1) + 1;
            words++;
        } while(iterator < sentence.length() && iterator != 0);
        
        return -1;
        
    }
}
